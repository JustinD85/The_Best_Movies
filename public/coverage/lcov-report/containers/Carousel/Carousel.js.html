<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/Carousel/Carousel.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">containers/Carousel</a> Carousel.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.92% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/11</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Fragment } from 'react'
import { Route, Switch } from 'react-router-dom'
import { Movie } from '../../components/'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import './Carousel.scss'
&nbsp;
export const Carousel = ({ movies, isPopup }) =&gt; {
  const movieCards = movies.map(movie =&gt; (
    &lt;Movie
      key={JSON.stringify(movie)}
      movie={movie} /&gt;))
  const favMovieCards = movieCards.filter(movieCard =&gt; movieCard.props.movie.isFavorite)
  const hasNoFavs = !favMovieCards.length &amp;&amp; &lt;h1 className='no-faves-notice'&gt;You have no favorites... yet!&lt;/h1&gt;
  const _class = isPopup ? 'Carousel blur' : 'Carousel'
&nbsp;
  return (
    &lt;Fragment&gt;
      &lt;section className={_class}&gt;
        &lt;Switch&gt;
          &lt;Route path='/favorites' render={<span class="fstat-no" title="function not covered" >()</span> =&gt; (
<span class="cstat-no" title="statement not covered" >            &lt;Fragment&gt;</span>
              &lt;h2 className='favorites-title'&gt;Favorite Movies&lt;/h2&gt;
              &lt;div className='movies-container'&gt;{(hasNoFavs || favMovieCards)}&lt;/div&gt;
            &lt;/Fragment&gt;)} /&gt;
&nbsp;
          &lt;Route path='/' render={<span class="fstat-no" title="function not covered" >()</span> =&gt; (
<span class="cstat-no" title="statement not covered" >            &lt;Fragment&gt;</span>
              &lt;h2 className='recents-title'&gt;Recent Movies&lt;/h2&gt;
              &lt;div className='movies-container'&gt;{movieCards}&lt;/div&gt;
            &lt;/Fragment&gt;)} /&gt;
        &lt;/Switch&gt;
      &lt;/section&gt;
    &lt;/Fragment&gt;
  )
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  movies: state.movies,
  isPopup: state.isPopup
})
&nbsp;
export default connect(mapStateToProps)(Carousel)
&nbsp;
Carousel.propTypes = {
  movies: PropTypes.array,
  isPopup: PropTypes.bool,
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 09:52:14 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
