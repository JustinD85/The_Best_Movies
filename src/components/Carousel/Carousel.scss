.Carousel {
  
}

.movie-title {
  color: #ECEBEE;
}