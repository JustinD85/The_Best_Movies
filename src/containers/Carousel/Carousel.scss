.Carousel {
  display: flex;
  flex-direction: column;
  padding: 6% 0% 0% 5%;
  color:#ECEBEE;
  
  .favorites-title,
  .recents-title {
    display: inline-block;
    text-align: left;
    padding-bottom: 2%;
  }
  
  .movies-container {
    display: flex;
    overflow: scroll;
    margin-right: 70px;
  }
}

.blur {
  filter: blur(3px)
}