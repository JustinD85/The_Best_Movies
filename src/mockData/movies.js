export const mockMovie = {
    "id": "5a7ee86cc3a368189300b429",
    "iso_639_1": "en",
    "iso_3166_1": "US",
    "key": "2mNMvz85NG8",
    "name": "The Grinch - In Theaters November 9 (TV Spot 1) (HD)",
    "site": "YouTube",
    "size": 720,
    "type": "Teaser",
}

export const mockMoviesResponse = {
    "id": 450465,
    "results": [
        {
            "id": "5a7ee86cc3a368189300b429",
            "iso_639_1": "en",
            "iso_3166_1": "US",
            "key": "2mNMvz85NG8",
            "name": "The Grinch - In Theaters November 9 (TV Spot 1) (HD)",
            "site": "YouTube",
            "size": 720,
            "type": "Teaser",
        }
    ]
}

export const mockMoviesArray = [
    {
        "id": "5a7ee86cc3a368189300b429",
        "iso_639_1": "en",
        "iso_3166_1": "US",
        "key": "2mNMvz85NG8",
        "name": "The Grinch - In Theaters November 9 (TV Spot 1) (HD)",
        "site": "YouTube",
        "size": 720,
        "type": "Teaser",
    }
]