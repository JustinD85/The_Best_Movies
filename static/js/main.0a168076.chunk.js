(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{28:function(e,t,n){e.exports=n(64)},33:function(e,t,n){},41:function(e,t,n){},44:function(e,t,n){},50:function(e,t,n){},52:function(e,t,n){},54:function(e,t,n){},56:function(e,t,n){},58:function(e,t,n){},60:function(e,t,n){},62:function(e,t,n){},64:function(e,t,n){"use strict";n.r(t);var r=n(0),a=n.n(r),o=n(26),c=n.n(o),s=(n(33),n(2)),i=n.n(s),u=n(5),l=n(10),p=n(11),m=n(15),f=n(12),v=n(16),h=n(46),d=n(68),E=n(7),b="https://falsemotive.io/api/",g=function(e,t){return{method:e,headers:{"Content-Type":"application/json"},body:JSON.stringify(t)}},w=function(){var e=Object(u.a)(i.a.mark(function e(){var t,n,r,a,o=arguments;return i.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:return t=o.length>0&&void 0!==o[0]?o[0]:"",n=o.length>1?o[1]:void 0,e.next=4,fetch("".concat(b).concat(t),g("POST",n));case 4:if((r=e.sent).ok){e.next=7;break}throw new Error("Could not post");case 7:return e.next=9,r.json();case 9:return a=e.sent,e.abrupt("return",a.data);case 11:case"end":return e.stop()}},e,this)}));return function(){return e.apply(this,arguments)}}(),O=function(){var e=Object(u.a)(i.a.mark(function e(){var t,n,r,a,o=arguments;return i.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:return t=o.length>0&&void 0!==o[0]?o[0]:"",n=o.length>1?o[1]:void 0,e.next=4,fetch("".concat(b).concat(t),g("DELETE",n));case 4:if((r=e.sent).ok){e.next=7;break}throw new Error("Could not delete");case 7:return e.next=9,r.json();case 9:return a=e.sent,e.abrupt("return",a.results);case 11:case"end":return e.stop()}},e,this)}));return function(){return e.apply(this,arguments)}}(),S=function(){var e=Object(u.a)(i.a.mark(function e(t){var n,r;return i.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("".concat(b).concat(t));case 2:if((n=e.sent).ok){e.next=5;break}throw new Error("".concat(n.statusText));case 5:return e.next=7,n.json();case 7:return r=e.sent,e.abrupt("return",r.data);case 9:case"end":return e.stop()}},e,this)}));return function(t){return e.apply(this,arguments)}}(),N=function(){var e=Object(u.a)(i.a.mark(function e(){var t,n;return i.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("https://falsemotive.io/newKey");case 2:return t=e.sent,e.next=5,t.json();case 5:return n=e.sent,e.abrupt("return",n);case 7:case"end":return e.stop()}},e,this)}));return function(){return e.apply(this,arguments)}}(),k=function(e){return{type:"SET_MOVIES",movies:e}},P=function(e,t){return{type:"SET_FAVORITES",favorites:e,user_id:t}},y=function(e){return{type:"SET_USER",user:e}},x=function(e){return{type:"SET_SHOULD_PROMPT_LOGIN",bool:e}},j=function(e){return{type:"SET_POPUP",bool:e}},C=n(43),_=(n(41),Object(E.b)(function(e){return{user:e.user}},function(e){return{setFavorites:function(t,n){return e(P(t,n))},setShouldPromptLogin:function(t){return e(x(t))},setPopup:function(t){return e(j(t))}}})(function(e){var t=e.movie,n=e.user,o=e.setShouldPromptLogin,c=e.setFavorites,s=e.isPopup,l=e.setPopup,p=function(){var e=Object(u.a)(i.a.mark(function e(){var r,a,s;return i.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,o(!n);case 2:if(!n){e.next=23;break}if(r="users/".concat(n.id,"/favorites"),a="".concat(r,"/").concat(t.id),e.prev=6,t.isFavorite){e.next=12;break}return e.next=10,w("/users/favorites/new",t);case 10:e.next=14;break;case 12:return e.next=14,O(a,t);case 14:return e.next=16,S(r);case 16:s=e.sent,c(s,n.id),e.next=23;break;case 20:e.prev=20,e.t0=e.catch(6),console.log(e.t0);case 23:case"end":return e.stop()}},e,this,[[6,20]])}));return function(){return e.apply(this,arguments)}}(),m=s?"popup":"Movie";return a.a.createElement("div",{className:m},s&&a.a.createElement(r.Fragment,null,a.a.createElement(C.a,{className:"close-popup",onClick:function(){return l(!1)},to:"/"},"X"),a.a.createElement("h1",{className:"movie-title"},t.title),a.a.createElement("p",{className:"movie-release"},"Released: ",t.release_date),a.a.createElement("p",{className:"movie-description"},t.overview),a.a.createElement("p",{className:"movie-vote"},"Average rating: ",t.vote_average),a.a.createElement("iframe",{width:"560",height:"315",title:t.title,src:"https://www.youtube.com/embed/".concat(t.trailer),frameBorder:"0",allow:"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",allowFullScreen:!0})),!s&&a.a.createElement(C.a,{to:"/movies/".concat(t.id)},a.a.createElement("img",{onClick:function(){return l(!0)},src:"https://image.tmdb.org/t/p/w500"+t.poster_path,className:"movie-poster",alt:"movie poster"})),a.a.createElement("span",{className:"favorite-btn",onClick:function(){return p()}},t.isFavorite?"\u2b50\ufe0f":"\u2606"))})),T=n(65),F=(n(44),function(){return a.a.createElement("nav",{className:"Nav"},a.a.createElement("h1",{className:"movie-tracker-title"},"Movie ",a.a.createElement("span",null,"Tracker")),a.a.createElement("div",{className:"nav-border"},a.a.createElement("div",{className:"nav-options"},a.a.createElement("p",{className:"browse"},"SORT BY:"),a.a.createElement(T.a,{exact:!0,to:"/",activeStyle:{color:"#0CD0FC"},name:"recent"},"RECENT"),a.a.createElement(T.a,{exact:!0,to:"/favorites",activeStyle:{color:"#0CD0FC"},name:"favorites"},"FAVORITES"))))}),L=(n(50),function(e){var t=e.movies;return a.a.createElement("div",{className:"Home"},a.a.createElement(F,null),a.a.createElement("section",{className:"View"},a.a.createElement(h.a,{path:"/",component:B}),a.a.createElement(h.a,{path:"/",render:function(e){var t=e.location;return a.a.createElement(R,{location:t})}}),a.a.createElement(h.a,{path:"/movies/:id",render:function(e){return function(e){var n=t.find(function(t){return t.id===parseInt(e.params.id)});return n?a.a.createElement(_,{key:JSON.stringify(n),movie:n,isPopup:!0}):a.a.createElement("div",null," Movie does not exist")}(e.match)}})))}),M=(n(52),function(e){var t=e.setShouldPromptLogin,n=e.setPopup;return n(!0),a.a.createElement("div",{className:"CreateAccountPrompt"},a.a.createElement("p",null,"Would you like to create an account?"),a.a.createElement(C.a,{to:"/create-user",onClick:function(){return t(!1)}},"YES"),a.a.createElement("button",{className:"no-btn",onClick:function(){t(!1),n(!1)}},"NO"))}),I=(n(54),function(e){function t(){var e,n;Object(l.a)(this,t);for(var r=arguments.length,a=new Array(r),o=0;o<r;o++)a[o]=arguments[o];return(n=Object(m.a)(this,(e=Object(f.a)(t)).call.apply(e,[this].concat(a)))).componentDidMount=Object(u.a)(i.a.mark(function e(){var t;return i.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:if(t=[],0!=localStorage.getItem("movies")){e.next=7;break}return e.next=4,N("now_playing");case 4:t=e.sent,e.next=8;break;case 7:t=JSON.parse(localStorage.getItem("movies"));case 8:n.props.setMovies(t);case 9:case"end":return e.stop()}},e,this)})),n}return Object(v.a)(t,e),Object(p.a)(t,[{key:"render",value:function(){var e=this,t=this.props,n=t.canPopup,r=t.setPopup,o=t.setShouldPromptLogin;return a.a.createElement("div",{className:"App"},a.a.createElement(h.a,{path:"/login",component:q}),a.a.createElement(h.a,{path:"/create-user",render:function(){return a.a.createElement(G,{setPopup:r})}}),a.a.createElement(h.a,{path:"/",render:function(){return n&&a.a.createElement(M,{setPopup:r,setShouldPromptLogin:o})}}),a.a.createElement(h.a,{path:"/",render:function(){return a.a.createElement(L,{movies:e.props.movies})}}))}}]),t}(r.Component)),U=Object(d.a)(Object(E.b)(function(e){return{movies:e.movies,canPopup:e.shouldPromptLogin,user:e.user}},function(e){return{setMovies:function(t){return e(k(t))},setShouldPromptLogin:function(t){return e(x(t))},setPopup:function(t){return e(j(t))}}})(I)),D=n(69),R=(n(56),Object(E.b)(function(e){return{movies:e.movies,isPopup:e.isPopup}})(function(e){var t=e.movies.map(function(e){return a.a.createElement(_,{key:JSON.stringify(e),movie:e})}),n=t.filter(function(e){return e.props.movie.isFavorite}),o=!n.length&&a.a.createElement("h1",{className:"no-faves-notice"},"You have no favorites... yet!"),c=e.isPopup?"Carousel blur":"Carousel";return a.a.createElement(r.Fragment,null,a.a.createElement("section",{className:c},a.a.createElement(D.a,null,a.a.createElement(h.a,{path:"/favorites",render:function(){return a.a.createElement(r.Fragment,null,a.a.createElement("h2",{className:"favorites-title"},"Favorite Movies"),a.a.createElement("div",{className:"movies-container"},o||n))}}),a.a.createElement(h.a,{path:"/",render:function(){return a.a.createElement(r.Fragment,null,a.a.createElement("h2",{className:"recents-title"},"Recent Movies"),a.a.createElement("div",{className:"movies-container"},t))}}))))})),A=n(14),V=n(66),G=(n(58),function(e){function t(){var e;return Object(l.a)(this,t),(e=Object(m.a)(this,Object(f.a)(t).call(this))).handleChange=function(t){var n=t.target,r=n.name,a=n.value;e.setState(Object(A.a)({},r,a))},e.handleSubmit=function(){var t=Object(u.a)(i.a.mark(function t(n){var r,a,o,c,s;return i.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return n.preventDefault(),r=e.state,a=r.name,o=r.email,c=r.password,s={name:a,email:o,password:c},t.prev=3,t.next=6,w("users/new",s);case 6:e.setState({didPost:!0}),t.next=12;break;case 9:t.prev=9,t.t0=t.catch(3),e.setState({error:"User already exists"});case 12:case"end":return t.stop()}},t,this,[[3,9]])}));return function(e){return t.apply(this,arguments)}}(),e.componentDidMount=function(){return e.props.setPopup(!0)},e.state={name:"",email:"",password:"",didPost:!1,error:""},e}return Object(v.a)(t,e),Object(p.a)(t,[{key:"render",value:function(){var e=this,t=this.state,n=t.didPost,r=t.error;return n?a.a.createElement(V.a,{to:"/login"}):a.a.createElement("div",{className:"CreateUser"},a.a.createElement("form",{onSubmit:this.handleSubmit},a.a.createElement("h1",{className:"create-your-account"},"Create Your Account"),a.a.createElement("input",{className:"user-input",placeholder:"Name",required:!0,type:"text",onChange:this.handleChange,name:"name"}),a.a.createElement("input",{className:"user-input",placeholder:"Email",required:!0,type:"email",onChange:this.handleChange,name:"email"}),a.a.createElement("input",{className:"user-input",placeholder:"Password",required:!0,type:"password",onChange:this.handleChange,name:"password"}),a.a.createElement("button",{className:"submit-user",type:"submit"},"Submit"),a.a.createElement(C.a,{to:"/",onClick:function(){return e.props.setPopup(!1)}},"Back"),r&&a.a.createElement("h3",null,r)))}}]),t}(r.Component)),J=(n(60),function(e){function t(){var e;return Object(l.a)(this,t),(e=Object(m.a)(this,Object(f.a)(t).call(this))).handleChange=function(t){var n=t.target,r=n.name,a=n.value;e.setState(Object(A.a)({},r,a))},e.setUser=Object(u.a)(i.a.mark(function t(){var n,r,a,o,c,s;return i.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return n=e.state,r=n.password,a=n.email,t.next=3,w("users",{password:r,email:a});case 3:return o=t.sent,c=o.name,s=o.id,e.props.setUser({name:c,id:s}),t.abrupt("return",s);case 7:case"end":return t.stop()}},t,this)})),e.setFavorites=function(){var t=Object(u.a)(i.a.mark(function t(n){var r,a;return i.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return r="users/".concat(n,"/favorites"),t.next=3,S(r);case 3:a=t.sent,e.props.setFavorites(a,n);case 5:case"end":return t.stop()}},t,this)}));return function(e){return t.apply(this,arguments)}}(),e.handleSubmit=function(){var t=Object(u.a)(i.a.mark(function t(n){var r;return i.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return n.preventDefault(),t.prev=1,t.next=4,e.setUser();case 4:r=t.sent,e.setFavorites(r),e.setState({canLogin:!0}),e.props.setPopup(!1),t.next=13;break;case 10:t.prev=10,t.t0=t.catch(1),e.setState({error:"That email or password does not exist"});case 13:case"end":return t.stop()}},t,this,[[1,10]])}));return function(e){return t.apply(this,arguments)}}(),e.componentDidMount=function(){e.props.setPopup(!0),e.props.setShouldPromptLogin(!1)},e.state={email:"",password:"",error:"",canLogin:!1},e}return Object(v.a)(t,e),Object(p.a)(t,[{key:"render",value:function(){var e=this,t=this.state,n=t.canLogin,r=t.error;return n?a.a.createElement(V.a,{to:"/"}):a.a.createElement("form",{className:"Login",onSubmit:this.handleSubmit},a.a.createElement("h1",null,"Movie Tracker"),a.a.createElement("h3",null,"Welcome back!"),a.a.createElement("input",{className:"user-input",placeholder:"Email",required:!0,type:"email",onChange:this.handleChange,name:"email"}),a.a.createElement("input",{className:"user-input",placeholder:"Password",required:!0,type:"password",onChange:this.handleChange,name:"password"}),r&&a.a.createElement("h3",null,r),a.a.createElement(C.a,{to:"/",onClick:function(){return e.props.setPopup(!1)}},"Back"),a.a.createElement("button",{className:"sign-in link"}," Sign In "),a.a.createElement("p",null,"New to Movie Tracker?"),a.a.createElement(T.a,{className:"create-user link",to:"/create-user"},"Create Account"))}}]),t}(r.Component)),q=Object(E.b)(function(e){return{movies:e.movies}},function(e){return{setMovies:function(t){return e(k(t))},setUser:function(t){return e(y(t))},setFavorites:function(t,n){return e(P(t,n))},setShouldPromptLogin:function(t){return e(x(t))},setPopup:function(t){return e(j(t))}}})(J),B=(n(62),Object(E.b)(function(e){return{user:e.user}},function(e){return{setUser:function(t){return e(y(t))},setMovies:function(t){return e(k(t))},setPopup:function(t){return e(j(t))}}})(function(e){var t=e.user,n=function(){var t=Object(u.a)(i.a.mark(function t(){return i.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return e.setUser(null),t.prev=1,t.t0=e,t.next=5,N("now_playing");case 5:t.t1=t.sent,t.t0.setMovies.call(t.t0,t.t1),t.next=12;break;case 9:t.prev=9,t.t2=t.catch(1),console.log("Error fetching data");case 12:case"end":return t.stop()}},t,this,[[1,9]])}));return function(){return t.apply(this,arguments)}}();return a.a.createElement(r.Fragment,null,t?a.a.createElement("div",{className:"Profile"},a.a.createElement("p",{className:"user-name"},t.name.toUpperCase()),a.a.createElement("button",{className:"logout-btn",onClick:function(){return n()}},"SIGN OUT")):a.a.createElement("div",{className:"Profile"},a.a.createElement(C.a,{to:"/login",className:"login-link",onClick:function(){return e.setPopup(!0)}},"SIGN IN")))})),X=n(67),Y=n(13),H=n(48),W=n(22),K=[],z=Object(Y.b)({movies:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:K,t=arguments.length>1?arguments[1]:void 0,n=t.type,r=t.id;switch(n){case"SET_MOVIES":return t.movies.map(function(e){return e.movie_id=e.id,e});case"TOGGLE_FAVORITE":return Object(W.a)(e).map(function(e){return e.id===r?Object(H.a)({},e,{isFavorite:!e.isFavorite}):e});case"SET_FAVORITES":return Object(W.a)(e).map(function(e){return e.isFavorite=!1,e.user_id=t.user_id,t.favorites.forEach(function(t){t.movie_id===e.movie_id&&(e.isFavorite=!0)}),e});default:return e}},user:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:null,t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_USER":return t.user;default:return e}},shouldPromptLogin:function(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_SHOULD_PROMPT_LOGIN":return t.bool;default:return e}},isPopup:function(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_POPUP":return t.bool;default:return e}}}),Q=window.__REDUX_DEVTOOLS_EXTENSION__&&window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(),Z=Object(Y.c)(z,Q),$=a.a.createElement(E.a,{store:Z},a.a.createElement(X.a,null,a.a.createElement(U,null)));c.a.render($,document.getElementById("root"))}},[[28,2,1]]]);
//# sourceMappingURL=main.0a168076.chunk.js.map