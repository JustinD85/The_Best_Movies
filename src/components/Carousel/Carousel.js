import React from 'react'
import Movie from '../../containers/Movie/Movie'
import { connect } from 'react-redux'
import './Carousel.scss'

const Carousel = (props) => {
  const movies = props.movies.map(movie => (
    <Movie movie={movie} />
  ))
  return <div className='Carousel'>{movies}</div>
} 

const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(Carousel)