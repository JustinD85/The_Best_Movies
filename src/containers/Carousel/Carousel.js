import React, { Fragment } from 'react'
import { Route, Switch } from 'react-router-dom'
import { Movie } from '../../components/'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import './Carousel.scss'

export const Carousel = ({ movies, isPopup }) => {
  const movieCards = movies.map(movie => (
    <Movie
      key={JSON.stringify(movie)}
      movie={movie} />))
  const favMovieCards = movieCards.filter(movieCard => movieCard.props.movie.isFavorite)
  const hasNoFavs = !favMovieCards.length && <h1 className='no-faves-notice'>You have no favorites... yet!</h1>
  const _class = isPopup ? 'Carousel blur' : 'Carousel'

  return (
    <Fragment>
      <section className={_class}>
        <Switch>
          <Route path='/favorites' render={() => (
            <Fragment>
              <h2 className='favorites-title'>Favorite Movies</h2>
              <div className='movies-container'>{(hasNoFavs || favMovieCards)}</div>
            </Fragment>)} />

          <Route path='/' render={() => (
            <Fragment>
              <h2 className='recents-title'>Recent Movies</h2>
              <div className='movies-container'>{movieCards}</div>
            </Fragment>)} />
        </Switch>
      </section>
    </Fragment>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies,
  isPopup: state.isPopup
})

export default connect(mapStateToProps)(Carousel)

Carousel.propTypes = {
  movies: PropTypes.array,
  isPopup: PropTypes.bool,
}