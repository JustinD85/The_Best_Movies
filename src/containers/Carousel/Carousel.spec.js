import React from 'react'
import { Carousel } from './Carousel'
import { shallow } from 'enzyme'
import { mockMoviesArray } from '../../mockData/movies.js'

describe('Carousel', () => {
  let wrapper
  it('should match the snapshot', () => {
    wrapper = shallow(<Carousel movies={mockMoviesArray} isPopup={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot', () => {
    wrapper = shallow(<Carousel movies={mockMoviesArray} />)
    expect(wrapper.find('.Carousel').exists()).toBe(true)
  })

  it('should match the snapshot', () => {
    wrapper = shallow(<Carousel movies={mockMoviesArray} isPopup={true} />)
    expect(wrapper.find('.blur').exists()).toBe(true)
  })
})
