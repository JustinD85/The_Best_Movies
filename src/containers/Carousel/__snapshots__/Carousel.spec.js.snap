// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match the snapshot 1`] = `
<Fragment>
  <section
    className="Carousel blur"
  >
    <Switch>
      <Route
        path="/favorites"
        render={[Function]}
      />
      <Route
        path="/"
        render={[Function]}
      />
    </Switch>
  </section>
</Fragment>
`;
